### DISCUSSION TITLE
Using Node JS Package with ROS2 data and CDR compression?

### DISCUSSION THREAD

**Discussion by gkjohnson** [foxglove/mcap#1178] [General]
Hello! I'm working from the `validate.ts` example referenced in https://github.com/foxglove/mcap/issues/585 and am trying to load a ROS bag data from my team but I'm getting the following error:

> ```
> Error: unsupported encoding cdr
> ```

I see that the [decompressHandlers](https://github.com/foxglove/mcap/blob/f4d29c6/typescript/support/src/decompressHandlers.ts#L4) function only provides handlers for zstd, lz4, and bzip. Are there packages, examples, or recommendations available for handling CDR compression?

**Comment by jtbandes** [comment]
Hey, sorry for the slow response here, this fell off my radar. CDR is a message encoding, not a type of compression. For specifically decoding ROS 2 messages we use https://github.com/foxglove/rosmsg2-serialization which is based on https://github.com/foxglove/cdr. Example here: https://github.com/foxglove/studio/blob/56620d28a684503a50f6c793b41b11e968b08254/packages/mcap-support/src/parseChannel.ts#L174

**Reply by jtbandes** [reply]
Fixing the `validate` example in https://github.com/foxglove/mcap/pull/1186

**Reply by gkjohnson** [reply]
Excellent - thank you!
